code/controllers/subsystem/tts.dm

```diff
@@ -0,0 +1,188 @@
+  *which determines the length of the timer for spam limiting and resets the HUD icon
+  *After the sound is played both files are deleted and the cycle begins anew
+  */
+
+#define GENERATOR_PATH    "tools\\tts_generator\\"	//TTS generator file location
+#define DATA_PATH         GENERATOR_PATH + "data\\"	//Temp files location
+#define STATUS_NEW        0
+#define STATUS_GENERATING 1
+#define STATUS_PLAYING    2
+
+SUBSYSTEM_DEF(tts)
+	name = "Text-to-Speech"
+	wait = 2
+	runlevels = RUNLEVEL_LOBBY | RUNLEVEL_GAME | RUNLEVEL_POSTGAME
+
+	/// List of items to process
+	var/list/processing
+
+/datum/controller/subsystem/tts/Initialize()
+	LAZYINITLIST(processing)
+
+	if (!CONFIG_GET(flag/enable_tts))
+		can_fire = FALSE
+	else
+		start_engine()
+
+	return ..()
+
+/**
+  *Launches the actual TTS generator
+  */
+/datum/controller/subsystem/tts/proc/start_engine()
+	if (!CONFIG_GET(flag/enable_tts))
+		return
+	var/cmd = "cmd /c start \"tts_generator\" [GENERATOR_PATH]tts_generator.exe"
+	shell(cmd)
+
+/datum/controller/subsystem/tts/proc/check_processing(client/C)
+	if (!C)
+		return FALSE
+
+	for (var/datum/tts/T in processing)
+		if (T.owner == C)
+			return TRUE
+
+	return FALSE
+
+/datum/controller/subsystem/tts/fire(resumed = FALSE)
+	if (!LAZYLEN(processing))
+		return
+
+	for (var/datum/tts/T in processing)	//we have something to process, let's make the files
+		switch(T.status)
+			if (STATUS_NEW)
+				var/uid = "[world.time]" + T.owner.ckey
+				fdel(DATA_PATH + "[uid].request")
+				fdel(DATA_PATH + "[uid].rlock")
+
+				text2file("", DATA_PATH + "[uid].rlock")
+				text2file("name=[uid]", DATA_PATH + "[uid].request")
+				text2file("voice=[T.voice]", DATA_PATH + "[uid].request")
+				text2file("text=[T.text]", DATA_PATH + "[uid].request")
+				fdel(DATA_PATH + "[uid].rlock")
+
+				T.filename = DATA_PATH + "[uid]"
+				T.status = STATUS_GENERATING
+				continue
+			if (STATUS_GENERATING)
+				// Check if this file is ready
+				if (fexists(T.filename + ".ogg") && fexists(T.filename + ".meta"))
+					play_tts(T)
+				continue
+			if (STATUS_PLAYING)
+				// Delete the file when it's finished
+				if (world.time > T.life)
+					delete_files(T)
+					LAZYREMOVE(processing, T)
+				continue
+/**
+  *Deletes files once they've been played
+  */
+/datum/controller/subsystem/tts/proc/delete_files(datum/tts/T)
+	if (!T)
+		return
+	if (!T.filename)
+		return
+	fdel(T.filename + ".ogg")
+	fdel(T.filename + ".meta")
+
+/datum/controller/subsystem/tts/proc/play_tts(datum/tts/T)
+	if (!T.owner)
+		message_admins("TTS request has no owner")	//I dunno how the fuck you managed to play a sound with no owner but don't
+		return
+	if (!T.owner.mob)
+		message_admins("TTS request has no mob")
+		return
+
+	var/next_channel = open_sound_channel()
+
+	T.status = STATUS_PLAYING
+
+	var/turf/origin = T.owner.mob.loc
+
+	var/list/listeners = GLOB.player_list
+	listeners = listeners & hearers(world.view, origin)

```

Comment:
There exists a game-mode-level player mob list, but is this operation necessary, given that you already check for `client` ahead to prune this list?